server/src/main/java/io/spine/server/bus/Bus.java

```diff
@@ -272,6 +329,57 @@ private void doPost(Iterable<E> envelopes, StreamObserver<IsSent> observer) {
         return messageConverter;
     }
 
+    public abstract static class AbstractBuilder<E extends MessageEnvelope<T>,
+            T extends Message,
+            B extends AbstractBuilder<E, T, B>> {
+
+        private final Queue<BusFilter<E>> filters;
```

Comment:
You have `Queue` here and `Deque` in the `FilterChain` below. Let's have it unified.